src/clusterfuzz/_internal/metrics/fuzzer_logs.py

```diff
@@ -102,21 +102,71 @@ def upload_to_logs(bucket_name,
 
   log_path = 'gs:/' + log_directory + '/' + get_log_relative_path(
       time, file_extension)
+  return log_path
+
+
+def upload_to_logs(bucket_name,
+                   contents,
+                   time=None,
+                   fuzzer_name=None,
+                   job_type=None,
+                   file_extension=None,
+                   signed_upload_url=None):
+  """Uploads file contents to log directory in GCS bucket.
+  Args:
+    bucket_name: Bucket logs are stored in.
+    contents: String containing log to be uploaded.
+    time: A datetime object used to generate filename for the log.
+    fuzzer_name: Name of the fuzzer. If None, gets this from the environment.
+    job_type: Job name. If None, gets this from the environment.
+    file_extension: A string appended to the end of the log filename. A default
+      value is used if None.
+    signed_upload_url: A signed url to upload the logs to.
+
+  Returns:
+    The path of the uploaded file and whether the uploaded succeeded.
+  """
+  if signed_upload_url:
+    if storage.upload_signed_url(contents, signed_upload_url):
```

Comment:
This seems sufficiently different enough to be its own function instead. None of the other arguments are used. Can we just change callers to call upload_signed_url directly instead?

Reply:
Great call. Done.